scripts/assign-colors.py

```diff
@@ -15,6 +25,7 @@
     parser.add_argument('--color-schemes', type=str, required=True, help="input color schemes file")
     parser.add_argument('--metadata', type=str, help="if provided, restrict colors to only those found in metadata")
     parser.add_argument('--clade-node-data', type=str, help="if provided, restrict to only those clades found in tree")
+    parser.add_argument('--clade-recency', type=int, help="if provided, restrict to clades found in tree within X months of present")
```

Comment:
_(non-blocking suggestion)_

How about encoding the unit somewhere in the option name or value? Examples:

1. `clade_recency_months: 6`
2. `clade_recency: 6M`

I'd lean towards (2) since a proper implementation would also allow different units e.g. `clade_recency: 1Y`.

If that sounds reasonable, I'd be happy to add a commit that does this – it'd be similar to https://github.com/nextstrain/augur/pull/740/commits/873a70a34105fd6579d2780d4dba946a1524363c

Reply:
Ah! Option 2 is a much nicer interface and would fit with usage of `min_date: "6M"`, etc... elsewhere in the config. If you're up for adding this commit that would be amazing.

Reply:
Added as 2bf39d32ceae17d13a73373314dc3f4c9b3bad68 and started a new [trial build](https://github.com/nextstrain/ncov/actions/runs/11113055010).

Reply:
Awesome. Thank you!

Reply:
Trial build did not add gray clades because I forgot an important part: 024ae9474ce79f584fa75efb70338250e1d00bf1

Started a new [trial build](https://github.com/nextstrain/ncov/actions/runs/11116193172).

Reply:
Hmm.... It looks like it's still not working quite correctly. https://nextstrain.org/staging/ncov/gisaid/trial/gray-out-early-clades-2/global/6m. Not exactly sure what went wrong.

Reply:
I noticed this too. Will investigate locally.

Reply:
🤦 I updated `assign-colors.py` but not the actual workflow. Updated with ea85954ffeb6eb77c962fd06f1b311380cdb26e6 and [re-running](https://github.com/nextstrain/ncov/actions/runs/11133100504) with the same trial name `gray-out-early-clades-2`.

Reply:
Still didn't work. I finally figured out how to test locally to check that `--clade-recency` is being passed through properly (should have done this much earlier):

```console
$ nextstrain build . --profile nextstrain_profiles/nextstrain-gisaid results/global_6m/colors.tsv --dry-run
...
        python3 scripts/assign-colors.py \
            --ordering defaults/color_ordering.tsv \
            --color-schemes defaults/color_schemes.tsv \
            --output results/global_6m/colors.tsv \
            --clade-node-data results/global_6m/clades.json \
            --clade-recency 6M \
            --metadata results/global_6m/metadata_adjusted.tsv.xz 2>&1 | tee logs/colors_global_6m.txt
...
```

Pushed the fix as 0b907ff3216ea8d76bb214dddc089a2ad1e2df41 and [re-running](https://github.com/nextstrain/ncov/actions/runs/11149144810) with the same trial name `gray-out-early-clades-2`.

Reply:
Thanks, noted for the future. The trial build completed and seems to be good: https://nextstrain.org/staging/ncov/gisaid/trial/gray-out-early-clades-2/global/6m